server/src/main/java/org/opensearch/search/profile/AbstractProfileBreakdown.java

```diff
@@ -45,44 +45,54 @@
  *
  * @opensearch.internal
  */
-public abstract class AbstractProfileBreakdown<T extends Enum<T>> {
+public abstract class AbstractProfileBreakdown {
 
-    /**
-     * The accumulated timings for this query node
-     */
-    protected final Timer[] timings;
-    protected final T[] timingTypes;
-    public static final String TIMING_TYPE_COUNT_SUFFIX = "_count";
-    public static final String TIMING_TYPE_START_TIME_SUFFIX = "_start_time";
+    public static final String NODE_TIME_RAW = "time_in_nanos";
+
+    private final Map<String, ProfileMetric> metrics;
 
     /** Sole constructor. */
-    public AbstractProfileBreakdown(Class<T> clazz) {
-        this.timingTypes = clazz.getEnumConstants();
-        timings = new Timer[timingTypes.length];
-        for (int i = 0; i < timings.length; ++i) {
-            timings[i] = new Timer();
+    public AbstractProfileBreakdown(Map<String, Class<? extends ProfileMetric>> metricClasses) {
+        Map<String, ProfileMetric> map = new HashMap<>();
+        for (Map.Entry<String, Class<? extends ProfileMetric>> entry : metricClasses.entrySet()) {
+            try {
+                map.put(entry.getKey(), entry.getValue().getConstructor(String.class).newInstance(entry.getKey()));
```

Comment:
We should not be using reflection here, some like `Collection<Supplier<ProfileMetric>>` should do the job, something along these lines (pseudo):

```
public AbstractProfileBreakdown(Collection<Supplier<ProfileMetric>> metricSuppliers) {
    metrics  = metricSuppliers.stream().map(Supplier::get).collectors(mapping(ProfileMetric::getName, metric);
}
```